tests/atgen/smsd-incoming-ussd.c

```diff
@@ -0,0 +1,67 @@
+#include <gammu-message.h>
+#include "test_helper.h"
+#include "../../include/gammu-smsd.h"
+#include "../../smsd/core.h"
+#include "../../libgammu/gsmstate.h"
+#include "../../smsd/services/files.h"
+
+GSM_Error SMSD_ConfigureLogging(GSM_SMSDConfig *Config, gboolean uselog);
+GSM_Error SMSD_SendSMS(GSM_SMSDConfig *Config);
+void SMSD_IncomingUSSDCallback(GSM_StateMachine *sm, GSM_USSDMessage *ussd, void *user_data);
+
+int main(void)
+{
+  GSM_Error error;
+  GSM_StateMachine *s = setup_state_machine();
+  GSM_Phone_ATGENData *Priv = setup_at_engine(s);
+  GSM_SMSDStatus status;
+  GSM_SMSDConfig *config = SMSD_NewConfig("test");
+  GSM_Protocol_Message msg;
+  const char *event = "+CUSD: 2";// USSD terminate for issue #464
+
+  const char *responses[] = {
+    // AT+CSCS=?
+    "+CSCS: (\"IRA\",\"GSM\",\"UCS2\")\r",
+    "OK\r\n",
+    // AT+CUSD=1
+    "OK\r\n",
+    // AT+CSCS="UCS2"
+    "OK\r\n",
+    // AT+CSCS?
+    "+CSCS:\r",
+    "OK\r\n",
+    // USSD DialService
+    "+CUSD: 0,\"0076006F00740072006500200072006500710075006500740065002000650073007400200065006E00200063006F0075007200730020006400650020007400720061006900740065006D0065006E0074002E00200076006500750069006C006C0065007A0020006C0069007200650020006C00650020006D00650073007300610067006500200070006C0075007300200074006100720064002E\",15\r\n",
+    "OK\r\n",
```

Comment:
Can you please also add testcase for the response only with code?

Reply:
Both cases are tested in the same function, line 20 is the incoming __+CUSD__ with only a code,  the test that is was received without error is lines 57-64. (lines 53-55 test the __+CUSD__ with all arguments)

The test is done like that during diagnostics to follow as closely as possible the pattern in the log attached to the issue, where a USSD request was made and 2 incoming USSD responses were received, the second of which was __+CUSD: 2__. I can extract it into its own testcase of you prefer?

Reply:
I don't mind having the test for complete response chain, it just didn't include the `+CUSD: 2` line, which is what you've fixed in the code.

Reply:
The test was there just easy to miss,  [line 20](https://github.com/gammu/gammu/blob/0bbafb481ead963ac00aa272119eed1612a29da6/tests/atgen/smsd-incoming-ussd.c#L20) declared the code only event and it was tested on [lines 57-64](https://github.com/gammu/gammu/blob/0bbafb481ead963ac00aa272119eed1612a29da6/tests/atgen/smsd-incoming-ussd.c#L57-L64).

It was a bit late and I decided it was fine to leave the debugging test, but it's a lot clearer now the test just addresses the change.